syweb/webclient/app/settings/settings-controller.js

```diff
@@ -264,25 +317,58 @@ function($scope, matrixService, modelService, eventHandlerService, notificationS
         setMuteNotifications(!$scope.config.muteNotifications);
     };
 
-    $scope.addContentRule = function(pattern, actions) {
-        notificationService.addGlobalContentRule(pattern, actions).then(function() {
+    $scope.addContentRule = function() {
+        var actions = [ $scope.rule_add_action.content ];
+        if ($scope.rule_add_sound.content) {
+            actions.push({'set_tweak': 'sound', 'value': 'default'});
+        }
+        actions.push({'set_tweak': 'highlight', 'value': $scope.rule_highlight.content});
```

Comment:
In a few places you inline JSON like this. That's okay for a single key/value pair, but when you start doing 2+ it isn't immediately obvious which ones are keys and which ones are values. It would be preferable to expand this out to:
```
actions.push({
  'set_tweak': 'highlight', 
  'value': $scope.rule_highlight.content
});
```
If you feel that this is going to inflate the # lines in this file too much, then it may be an indication that this logic should be split out into a helper ``Service`` of some kind, which can more easily map between button clicks and the underlying JSON that gets added.